src/cart/controller/cart.controller.ts

```diff
@@ -0,0 +1,44 @@
+import { Controller, Post, Get, Body, UseGuards, Patch } from '@nestjs/common';
+
+import { User } from '~/users/schema/users.schema';
+import { JwtAuthGuard } from '~/auth/guards/auth.guard';
+import { CurrentUser } from '~/auth/decorators/current-user.decorator';
+import { Role } from '~/auth/role.enum';
+import { Roles } from '~/auth/decorators/roles.decorator';
+import { RolesGuard } from '~/auth/guards/role.guard';
+
+import { AddToCartDto } from '../dto/addToCart.dto';
+import { CartService } from '../service/cart.service';
+import { RemoveFromCartDto } from '../dto/removeFromCart.dto';
+
+@Controller('carts')
+export class CartController {
+    constructor(private readonly cartService: CartService) { }
+
+    @Post()
+    @Roles(Role.USER)
+    @UseGuards(JwtAuthGuard, RolesGuard)
```

Comment:
I think we can scope this to the controller instead of the individual methods.

Reply:
okay , I didn't know nest was capable of that